numpy/polynomial/chebyshev.py

```diff
@@ -1886,6 +1887,59 @@ def chebroots(c):
     return r
 
 
+def chebinterp(func, deg, domain=None, args=()):
+    """Return the Chebyshev series instance that interpolates a given function
+    at the Chebyshev points of the first kind (mapped to the given domain).
+    The function is interpolated at the Chebyshev points of the first kind,
+    mapped onto the domain of the function being approximated.
+
+    Parameters
+    ----------
+    func : function
+        The function to be approximated. It must be a function of a single
+        variable of the form f(x,a,b,c...), where a,b,c... are extra arguments
+        that can be passed in the `args` parameter.
+    deg : int or 1-D array_like
+        Degree(s) of the fitting polynomials. If `deg` is a single
```

Comment:
The other option is to simply omit the domain and define it as an interpolation of a function over the interval [-1, 1], which is what I prefer. The classmethod then calls it with `lambda x: func(mapdomain(x, [-1, 1], domain), *args)`. The virtue of that is that the returned Chebyshev instance knows how to handle the mapping between domain and window transparently.